src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -269,6 +287,37 @@ internal static int NumberOfValueTuples(int numElements, out int remainder)
         /// <summary>
         /// Produces the underlying ValueTuple corresponding to this list of element types.
         /// </summary>
+        private static NamedTypeSymbol GetTupleUnderlyingType(ImmutableArray<TypeSymbol> elementTypes, CSharpCompilation compilation)
```

Comment:
I was not too happy about this duplication either. Thanks for the tips for how to do it.
One other trouble I have here is that the original method needed a syntax node (to get a location in reported diagnostics), but in the new scenario I have no node.